syntax = "proto3";

package bot_builder_story.service;

service BotBuilderStoryService {
  rpc GetStoryBlocks(GetStoryBlocksRequest) returns (StoryBlock);
  rpc GetBotResponses(GetBotResponsesRequest) returns (GetBotResponsesResponse);
}

message GetStoryBlocksRequest {
  string user_id = 1;
}

message StoryBlock {
  string id = 1;
  string name = 2;
  string type = 3;
  string user_id = 4;
  string parent_id = 5;
  repeated StoryBlock children = 6;
}

message GetBotResponsesRequest {
  string story_block_id = 2;
}

message GetBotResponsesResponse {
  repeated BotResponse responses = 1;
}

message BotResponse {
  string type = 1;
  repeated string variants = 2;
}